### PR TITLE
Validations and model specs

### PR DESCRIPTION
**Pull request by Smart1-hub**
Add the following validations:
    For the User model:
          Name must not be blank.
          PostsCounter must be an integer greater than or equal to zero.
    For the Post model:
          Title must not be blank.
          Title must not exceed 250 characters.
          CommentsCounter must be an integer greater than or equal to zero.
          LikesCounter must be an integer greater than or equal to zero.    
Add unit specs for all of your models' methods and validations.

### CODE REVIEW

**Review by rotimiazeez — CHANGES_REQUESTED** [review]
# Hi @Smart1-hub 👏🏽, 
## Good job so far! 👍🏽 
### Summary ♨️ 
- No Linter error ✔️ 
- Good and descriptive PR 💯
- Good commit messages 📖
There are some issues that you still need to work on to go to the next project but you are almost there!
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>

## Highlights


## Required Changes ♻️
### Kindly make sure you implement unit tests for `User`, `Post`, `Comment` and `Like` model

- Model specs have **Validation** tests Example:

  ```ruby
  describe 'validations' do
    it { should validate_presence_of(:name) }
  end
  ```
- Any custom methods in the models should have a corresponding unit test.
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Smart1-hub — COMMENTED** [review]
Add unit tests for model as required by the reviewer

File: spec/models/like_spec.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+require 'rails_helper'
+RSpec.describe Like, type: :model do
```
Comment (FrederickMih) [inline spec/models/like_spec.rb:2]: - [ ] Hi @Smart1-hub, kindly add tests for your Like model. Let's add validation tests and equally test the `update_likes_counter` method.

File: spec/models/post_spec.rb [hunk 2]
```diff
@@ -0,0 +1,30 @@
+require 'rails_helper'
+
+RSpec.describe Post, type: :model do
+  user = User.create(name: 'Tom', bio: 'teacher in uganda')
```
Comment (FrederickMih) [inline spec/models/post_spec.rb:4]: - [ ] I got this console error when I ran your tests. Kindly fix it.

Screenshot from 2022-08-24 16-05-59

**Review by FrederickMih — CHANGES_REQUESTED** [review]
Hi @Smart1-hub , 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- [ ] Added unit tests for User, Post, and comment model :heavy_check_mark: 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Smart1-hub — COMMENTED** [review]
Update reviewer's comments

**Review by mikethreels — APPROVED** [review]
Hi Ademola, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**